### ISSUE TITLE
GameModeInventories Crash Report

### ISSUE THREAD

**Issue by kathfishing** [eccentricdevotion/GameModeInventories#16]
Hi there - Our server has crashed 17 times in the past 3 days to what we believe is an issue with GameModeInventories based on every crash report.  We are not receiving any console errors and are having a hard time figuring out the cause of this.

[GameModeInventories Server Crash_.zip](https://github.com/eccentricdevotion/GameModeInventories/files/8203088/GameModeInventories.Server.Crash_.zip)

If someone could please take a look and try to help us determine the cause and how to fix this, we would greatly appreciate it.  We are dependent upon GameModeInventories for a very important aspect of our server.

We are currently running 1.18.1 with the latest version of GameModeInventories.

Thank you so much for your help!

**Comment by eccentricdevotion** [comment]
There is nothing in the zip file except a `.code-workspace` file that contains no relevant information

**Comment by GroundedMC** [comment]
> There is nothing in the zip file except a `.code-workspace` file that contains no relevant information

Oh my word - sorry about that!

[crash-2022-03-08_05.30.09-server.txt](https://github.com/eccentricdevotion/GameModeInventories/files/8209953/crash-2022-03-08_05.30.09-server.txt)

**Comment by eccentricdevotion** [comment]
`OutOfMemoryError`

How big is your GMI.db database? Looks like there are millions of blocks recorded as I presume you are using the plugin as some sort of creative gamemode blocker and have `track_creative_place` enabled. While the plugin has this function, it was only really designed as an inventory management tool - looks like the load is now too big for the server to handle.

Attach/link to database file?

**Comment by GroundedMC** [comment]
> `OutOfMemoryError`
> 
> How big is your GMI.db database? Looks like there are millions of blocks recorded as I presume you are using the plugin as some sort of creative gamemode blocker and have `track_creative_place` enabled. While the plugin has this function, it was only really designed as an inventory management tool - looks like the load is now too big for the server to handle.
> 
> Attach/link to database file?

Yeah, we are using it for a creative gamemode within a survival eco server and do have 'track_creative_place' enabled.  However, it looks like we had it set to sqlite instead of mysql.. could that be the issue?
https://i.imgur.com/ObEACuN.png

Our GMI.db file is 2.75mb. 
[GMI.zip](https://github.com/eccentricdevotion/GameModeInventories/files/8210074/GMI.zip)

**Comment by GroundedMC** [comment]
> The database doesn't look so big, there are only 15,000 block records - I would have thought the server could handle that fine, especially as the plugin only loads blocks for loaded chunks as necessary.
> 
> Yes, I would try switching to MySQL (on a backup test server first), though you will need to convert your sqlite database over.

Ok, thanks, I will be in touch after we test that out and figure out how to convert the database to mysql.  Thanks for your help!

**Comment by GroundedJDU** [comment]
[crash-2022-04-05_22.40.33-server.txt](https://github.com/eccentricdevotion/GameModeInventories/files/8447167/crash-2022-04-05_22.40.33-server.txt)

Hello.. we switched to mySQL database and everything was working fine until yesterday when we started losing memory again. We didnt have a lot of players using gamemode creative so i dont think the database is so big that the server cant handle it.

I attached the latest crash report and our config.. if you could check them out that would be great.
Thanks.

[Config.txt](https://github.com/eccentricdevotion/GameModeInventories/files/8447284/Config.txt)

**Comment by eccentricdevotion** [comment]
Hikari connection pool can't get a connection to MySQL

**Comment by GroundedJDU** [comment]
I dont really know what Hikari connection pool is.. and is there a fix for our issue? do we just not use mysql for gamemode inventories?

**Comment by eccentricdevotion** [comment]
You should be able to use MySQL for GMI, I'm not sure what your issue is, but it looks like the MySQL server is not responding? Do you have any other plugins that use MySQL?

**Comment by GroundedJDU** [comment]
Yeah, we have a bunch and they all work fine.

**Comment by eccentricdevotion** [comment]
Try lowering your connection pool. size in the GMI config:

```
storage:
  mysql:
    pool_size: 5
```

The error in the crash report is references Hikari connection pool a third party library that a lot of plugins use to connect to MySQL, as such it is not my code so there is little/nothing I can do to change it.

I'll keep looking into it.

**Comment by eccentricdevotion** [comment]
If the `pool_size` doesn't work at `5` try going the other way and making it larger `pool_size: 100` for example

**Comment by eccentricdevotion** [comment]
Could also try the latest build...
http://tardisjenkins.duckdns.org:8080/job/GameModeInventories/

**Comment by GroundedJDU** [comment]
The latest build still crashes the server. I also lowered pool_size to 5.. still crashed.

**Comment by eccentricdevotion** [comment]
What about increasing the pool size as mentioned in my previous comment?

**Comment by GroundedJDU** [comment]
Tried increasing it too, still crashes every time i join the server. It does run while no one is on.. but as soon as someone joins the server it crashes.

**Comment by eccentricdevotion** [comment]
Made some small changes which should make the plugin slightly more efficient, try the latest build.

Also tested with 16,000+ block records with both MySQL and SQLite and not seeing any issues with crashes, out-of-memory errors or Hikari connection pool issues.

**Comment by GroundedJDU** [comment]
Hey.. sorry about late response, I wanted to try some other things before I do, but the last update didnt help. Can you please send me your config so i can compare them next to each other and see if I did something wrong in ours? And is there any way to limit all containers without that big list in config? 
Thanks.

**Comment by GroundedJDU** [comment]
Also, is there any way to disable armor stand interaction?
